docs/latest/examples/authentication-with-supabase.md

```diff
@@ -0,0 +1,263 @@
+on the topic.
+
+The purpose of the example app we're building here is to showcase the basic
+building blocks of an implementation. As such, it is limited in functionality
+and purposefully leaves out things like
+[password resets](https://supabase.com/docs/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr),
+[proper error handling](https://fresh.deno.dev/docs/concepts/error-pages) as
+well as validating input form data. You can find the
+[full code here](https://github.com/morlinbrot/supa-fresh-pkce), where the
+missing functionality is implemented.
+
+## Supabase
+
+First of all, we need a Supabase account
+[which can be created for free here](https://supabase.com/). A handy way to
+supply the credentials to our app is via `.env` file (never check in `.env`
+files to version control).
+
+```txt .env.example
+SUPABASE_URL=https://<projectName>.supabase.co
+SUPABASE_ANON_KEY=<api_key>
+```
+
+Update the imports section of your `deno.json` file to include the following:
+
+```json deno.json
+"imports": {
+  "supabase": "npm:@supabase/supabase-js@2",
+  "supabase/ssr": "npm:@supabase/ssr",
+}
+```
+
+Since Deno 1.38, we reading .env files is built-in and can be enabled with the
+`--env` flag. Here's the complete command to run our app:
+
+```shell
+deno run --unstable-kv --allow-env --allow-read --allow-write --allow-run --allow-net --watch=static/,routes/ dev.ts
+```
+
+### `@supabase/ssr`
+
+Supabase provides the `@supabase/ssr` package for working with its API in an SSR
+context. It exposes the `createServerClient` method that we can use on the
+server side. Set it up like so:
+
+```ts lib/supabase.ts
+import { deleteCookie, getCookies, setCookie } from "$std/http/cookie.ts";
+import { assert } from "$std/assert/assert.ts";
+import { type CookieOptions, createServerClient } from "supabase/ssr";
+
+export function createSupabaseClient(
+  req: Request,
+  // Keep this optional parameter in mind, we'll get back to it.
+  resHeaders = new Headers(),
+) {
+  const SUPABASE_URL = Deno.env.get("SUPABASE_URL");
+  const SUPABASE_ANON_KEY = Deno.env.get("SUPABASE_ANON_KEY");
+
+  assert(
+    SUPABASE_URL && SUPABASE_ANON_KEY,
+    "SUPABASE URL and SUPABASE_ANON_KEY environment variables must be set.",
+  );
+
+  return createServerClient(SUPABASE_URL, SUPABASE_ANON_KEY, {
+    auth: { flowType: "pkce" },
+    cookies: {
+      get(name: string) {
+        return decodeURIComponent(getCookies(req.headers)[name]);
+      },
+      set(name: string, value: string, options: CookieOptions) {
+        setCookie(resHeaders, {
+          name,
+          value: encodeURIComponent(value),
+          ...options,
+        });
+      },
+      remove(name: string, options: CookieOptions) {
+        deleteCookie(resHeaders, name, options);
+      },
+    },
+  });
+}
+```
+
+Note: We are specifying the `flowType` to be `pkce` and that we're using
+`encodeURIComponent` to serialize and store the session object as a cookie.
```

Comment:
```suggestion
> Note: We are specifying the `flowType` to be `pkce` and that we're using
> [`encodeURIComponent()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) to serialize and store the session object as a cookie.
```

Reply:
This was in the first review, right? Must've overlooked this. Anyways, it's fixed now.